Firebase Android Integration

Could you update the firbase integration for Andriod. Trying to get it to run threw up numerous errors.

after fixing dependency issues I am now left with:

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:processDebugGoogleServices'.
> No matching client found for package name 'de.janoodle.circlesApp.debug'

```

Obviously made sure the package names match. Sure the firebase integration/Android integration is as straigh forward as the iOS one?

What's your app id on Firebase?

On Fri, Oct 11, 2019, 12:03 bkvie <notifications@github.com> wrote:

> after fixing dependency issues I am now left with:
>
> FAILURE: Build failed with an exception.
>
> * What went wrong:
> Execution failed for task ':app:processDebugGoogleServices'.
> > No matching client found for package name 'de.janoodle.circlesApp.debug'
>
>
> Obviously made sure the package names match. Sure the firebase
> integration/Android integration is as straigh forward as the iOS one?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/janoodleFTW/timy-messenger/issues/16?email_source=notifications&email_token=AATA7KBBENOJL6SBRN65HALQOBFN7A5CNFSM4I7WI2RKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEA7QTAI#issuecomment-541002113>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AATA7KCLR6OIUTKPUMDLXUDQOBFN7ANCNFSM4I7WI2RA>
> .
>

--- a/android/README.md
+++ b/android/README.md
@@ -1,11 +1,35 @@
 # Timy Android
 
-## Firebase
+## Setting up Firebase
 
-The Firebase configuration file `google-services.json` need to be provided and located inside the `android/src/main` folder.
+Create a project on the Firebase console [here](https://console.firebase.google.com/)
 
-If you want to have different configs for release and debug, 
-provide two different files in the `android/src/release` and `android/src/debug` folders.
+- To add Firebase to your app, click on the android icon or click the gear icon to go to project
+settings to find the android icon.
+
+- Register your application by filing up the form with the package name (applicationId) 
+and the app nickname if you like.
+> Find Your package name which is generally the applicationId in your app-level build.gradle file
+
+- Don't just download the `google-service.json` just yet, click next throughout and also skip the
+step where you need to verify the configuration.
+
+- Go back to project settings, here we need to generate a second configuration file. The reason for
+this is that the first file we generated is for a release build. Now we need to create one for a
+debug build. These applications are differentiated by the application id to which while debugging,
+a suffix `.debug` is appended to the applicationId.
+
+- Under your apps, add another app to the project. Click the android icon again on the dialog that
+pops up. Register your app with the package name with `.debug` at the end.
+
+- Download the `google-service.json` file that is generated for you. Find it and move it inside
+the folder `android/app/` of the project. The firebase sdk is already added to the project.
+
+- Open and view the contents of the file, check that there are 2 entries of `client_info` with the
+2 package names, one for debug and one for release.
+
+- On the fourth step of registration, run the app to verify the configuration via the Firebase
+console.
 
 ## Distribution
 

